Backpack/Snackbar/Classes/BPKSnackbar.m

```diff
@@ -0,0 +1,370 @@
+                      delegate:(id<BPKSnackbarProtocol> _Nullable)delegate {
+    self.titleLabel.text = title;
+    self.delegate = delegate;
+    self.snackbarView.backgroundColor = BPKColor.skyBlueShade02;
+    self.parentViewController = viewController;
+    self.actionButton.title = buttonTitle;
+    self.actionButton.image = buttonIcon;
+    self.leftIconContainer.image = leftIcon;
+
+    if (buttonTitle != nil || buttonIcon != nil) {
+        self.actionButton.hidden = NO;
+        self.labelButtonSpacing.constant = BPKSpacingLg;
+        [self.actionButton setContentCompressionResistancePriority:751 forAxis:UILayoutConstraintAxisHorizontal];
+    } else {
+        self.actionButton.hidden = YES;
+        self.labelButtonSpacing.constant = 0;
+        [self.actionButton setContentCompressionResistancePriority:UILayoutPriorityDefaultHigh
+                                                           forAxis:UILayoutConstraintAxisHorizontal];
+    }
+
+    if (leftIcon != nil) {
+        self.leftIconContainer.hidden = NO;
+        self.labelLeadingSpacing.constant = BPKSpacingMd;
+    } else {
+        self.leftIconContainer.hidden = YES;
+        self.labelLeadingSpacing.constant = 0;
+    }
+
+    if (duration != BPKSnackbarDurationIndefinite) {
+        self.durationInMillis = [self durationInMilliseconds:duration];
+        if (self.durationInMillis > 0) {
+            self.timer = [NSTimer scheduledTimerWithTimeInterval:self.durationInMillis / 1000.0
+                                                          target:self
+                                                        selector:@selector(snackbarTimeOut)
+                                                        userInfo:nil
+                                                         repeats:NO];
+        }
+    }
+
+    [self updateLayout];
+}
+
+- (void)updateLayout {
+    UIView *presentingView = self.parentViewController.view;
+    [presentingView addSubview:self];
+
+    if (self.superview) {
+        [NSLayoutConstraint activateConstraints:@[
+            [self.leftAnchor constraintEqualToAnchor:self.superview.leftAnchor],
+            [self.rightAnchor constraintEqualToAnchor:self.superview.rightAnchor]
+        ]];
+
+        NSLayoutYAxisAnchor *bottomAnchor = self.superview.safeAreaLayoutGuide.bottomAnchor;
+
+        self.bottomConstraint =
+            [self.bottomAnchor constraintEqualToAnchor:bottomAnchor
+                                              constant:[self bottomInsetWithKeyboardHeight:self.keyboardHeight]];
+        self.bottomConstraint.active = YES;
+
+        self.heightConstraint = [self.heightAnchor constraintEqualToConstant:BPKSnackbarHeight];
+        self.heightConstraint.active = YES;
+
+        [self layoutIfNeeded];
+    }
+}
+
+- (void)show {
+    // First check if there's another snackbar presented
+    [self findAndRemoveSnackbars];
+
+    CGFloat distance = self.heightConstraint.constant;
```

Comment:
Shouldn't this wait for the previous snackbar to animate out if there is one?

Reply:
Is not necessary, the transition is very smooth, here's a gif with slow motion
![ezgif com-video-to-gif-2](https://user-images.githubusercontent.com/10763919/69265439-2d033500-0bc1-11ea-9050-d7f224d8ee95.gif)